app/controllers/materials_controller.rb

```diff
@@ -11,9 +24,8 @@ def index
 
   def show
     @material = Material.find(params[:id])
-    @attributes = @material.attributes.except(
-      "name", "manufacturer_id", "created_at", "updated_at"
-    )
+    @attributes = @material.attributes.slice(
+      *MATERIAL_ATTRIBUTES["default"]+MATERIAL_ATTRIBUTES[@material.attributes["function"]])
```

Comment:
Might be nice to move this to a view helper as it doesn't really belong in the controller. I should have done this with the original code.